File: tests/rspec/pages/members/profile/username_password.rb
require_relative '_profile_form.rb'

module Profile
  class UsernamePassword < ControlPanelPage
    set_url '/admin.php?/cp/members/profile/auth'

    section :profile_form,
      Profile::ProfileForm,
      'form[action*="cp/members/profile"]'

    element :username, 'input[name=username]'
    element :screen_name, 'input[name=screen_name]'
    element :password, 'input[name=password]'
    element :confirm_password, 'input[name=confirm_password]'
    element :current_password, 'input[name=verify_password]'
  end
end

File: tests/cypress/cypress/elements/pages/members/profile/UsernamePassword.js
import ProfileForm from './ProfileForm'

class UsernamePassword extends ProfileForm {
  constructor() {
      super()
      this.url = '/admin.php?/cp/members/profile/auth';

      this.elements({

        'profile_form': 'form[action*="cp/members/profile"]',
        'profile_form_submit': 'form[action*="cp/members/profile"] div.form-btns.form-btns-top input[type=submit]',

        'username': 'input[name=username]:visible',
        'screen_name': 'input[name=screen_name]:visible',
        'password': 'input[name=password]:visible',
        'confirm_password': 'input[name=confirm_password]:visible',
        'current_password': 'input[name=verify_password]:visible'

      })
  }
}
export default UsernamePassword;
